### PR TITLE
React中的State是如何工作的——用代码示例解释

### LINKED ISSUE #1002: [Auto]（React中的State是如何工作的——用代码示例解释）
- 原文网址：[How State Works in React – Explained with Code Examples](https://www.freecodecamp.org/news/what-is-state-in-react-explained-with-examples/)
- MarkDown 文件：https://github.com/freeCodeCamp/news-translation/edit/main/chinese/articles/what-is-state-in-react-explained-with-examples.md

### PR DESCRIPTION
**Pull request by dake0913**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #1002

### CODE REVIEW

**Comment by dake0913** [comment]
https://github.com/dake0913/news-translation/blob/patch-8/chinese/articles/what-is-state-in-react-explained-with-examples.md
@miyaliu666 已完成翻译，请校对

File: chinese/articles/what-is-state-in-react-explained-with-examples.md [hunk 1]
```diff
@@ -723,47 +711,46 @@ const User = (props) => {
...
 
-> React hooks are functional components with added state and lifecycle methods.
+>React钩子是添加了状态和生命周期方法的函数组件。
+所以现在,类组件和函数组件之间没有什么区别。
 
-So now, there is very little to no difference between class-based components and functional components.
+它们都可以有状态和生命周期方法。
 
-Both of them can have state and life cycle methods.
+但是,反应钩子现在喜欢写反应组件,因为它们使代码更短,更容易理解。
```
Comment (miyaliu666) [inline chinese/articles/what-is-state-in-react-explained-with-examples.md:741]: 反应钩子现在喜欢写反应组件 => 在写 React 组件时，React hook 更常用

File: chinese/articles/what-is-state-in-react-explained-with-examples.md [hunk 2]
```diff
@@ -663,27 +650,27 @@ this.setState((prevState) => ({
...
+## 如何在React的函数组件中使用状态
 
-Up until now, we've seen how to use state in class-based components. Let's now see how to use it in functional components.
+到目前为止，我们已经了解了如何在类组件中使用状态。现在让我们看看如何在函数组件中使用它。
 
-Functional components are similar to class components, except that they do not have state and lifecycle methods. This is why you may have heard them called stateless functional components.
+函数组件与类组件类似，只是它们没有状态和生命周期方法。这就是为什么您可能听说过它们被称为无状态函数组件。
 
-These components only accept props and return some JSX.
+这些组件只接受道具并返回一些JSX。
```
Comment (miyaliu666) [inline chinese/articles/what-is-state-in-react-explained-with-examples.md:665]: props 不翻译

File: chinese/articles/what-is-state-in-react-explained-with-examples.md [hunk 3]
```diff
@@ -817,31 +804,29 @@ const App = () => {
...
+* 注意，因为我们已经有了计数器的值，所以我们使用 `setCounter(counter + 1)` 来增加计数器的值。
+* 由于内联on click处理程序中只有一条语句，所以不需要将代码移动到单独的函数中。不过，如果处理程序内部的代码变得复杂，您可以这样做。
+如果您想了解关于useState和其他React hook的更多细节(以及示例)，请查看我的React hook介绍文章。
 
 ### Thanks for reading!
 
-Want to learn all ES6+ features in detail including let and const, promises, various promise methods, array and object destructuring, arrow functions, async/await, import and export and a whole lot more from scratch?
-
-**Check out my [Mastering Modern JavaScript](https://modernjavascript.yogeshchavan.dev/) book. This book covers all the pre-requisites for learning React and helps you to become better at JavaScript and React.**
+想要详细了解所有ES6+特性，包括let和const，承诺，各种承诺方法，数组和对象解构，箭头函数，异步/等待，导入和导出，以及从头开始的很多更多?
```
Comment (miyaliu666) [inline chinese/articles/what-is-state-in-react-explained-with-examples.md:823]: promises 不翻译

Comment (miyaliu666) [inline chinese/articles/what-is-state-in-react-explained-with-examples.md:823]: async/await 也不翻译

**Review by miyaliu666 — COMMENTED** [review]
谢谢 @dake0913 翻译这篇文章，译文整体比较流畅，我在发布时修改几个错误：）

**Pull request merged**